File: 448-find-all-numbers-disappeared-in-an-array/448-find-all-numbers-disappeared-in-an-array.cpp
class Solution {
public:
    vector<int> findDisappearedNumbers(vector<int>& nums)
    {
      for (int i = 0; i < nums.size(); i++) 
            nums[abs(nums[i])-1] = -abs(nums[abs(nums[i])-1]);
        
        vector<int> res;
        for(int a:nums) cout<<a<<" ";
        for (int i = 0; i < nums.size(); i++)
            if (nums[i] > 0)
                res.push_back(i+1);
        
        return res;
          
    }
};

File: 448-find-all-numbers-disappeared-in-an-array/448-find-all-numbers-disappeared-in-an-array.java
class Solution {
    public List<Integer> findDisappearedNumbers(int[] nums) {
        for (int i = 0; i < nums.length; i++) 
            nums[Math.abs(nums[i])-1] = -Math.abs(nums[Math.abs(nums[i])-1]);
        
        List<Integer> res=new ArrayList<>();
        //for(int a:nums) cout<<a<<" ";
        for (int i = 0; i < nums.length; i++)
            if (nums[i] > 0)
                res.add(i+1);
        
        return res;
    }
}